### PR TITLE
fix(admin): display initial loading on managers and mempership pages …

### PR DESCRIPTION
**Pull request by michalberky**
…in vo settings

* Added spinner for vo-settings-expiration component
* Fixed spinners in managers-page component

### CODE REVIEW

File: apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-expiration/vo-settings-expiration.component.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<mat-spinner *ngIf="expirationAttribute === undefined" class="ms-auto me-auto"> </mat-spinner>
```
Comment (xflord) [inline apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-expiration/vo-settings-expiration.component.html:1]: the perun loading directive and a loading field which would also work for when saving changes seems more fitting to me

**Comment by michalberky** [comment]
I am not sure if the spinner taking up this whole area is ok, but I couldn't find any other solution that wouldn't require huge changes. So please let me know if there is any or it needs to be changed either way

File: apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-expiration/vo-settings-expiration.component.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export class VoSettingsExpirationComponent implements OnInit {
   @HostBinding('class.router-component') true;
 
   expirationAttribute: Attribute;
+  loading = false;
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-expiration/vo-settings-expiration.component.ts:23]:
```suggestion
  expirationAttribute: Attribute = { id: -1, beanName: 'Attribute', value: null };
  loading = false;
```
For the correct behavior, we need to add in `expiratoin-settings.component.ts` this update too:
```
this.currentConfiguration = this.unParseAttrValue(
   this.expirationAttribute.value as ExpirationAttrValue,
);
```
And maybe we can remove this and also initialConfiguration setting in ngOnInit here.

File: apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-expiration/vo-settings-expiration.component.html [hunk 3]
```diff
@@ -1,5 +1,10 @@
...
+<ng-template #spinner>
+  <perun-web-apps-loading-table></perun-web-apps-loading-table>
+</ng-template>
+<ng-container *ngIf="expirationAttribute !== undefined; else spinner">
+  <app-expiration-settings
+    *perunWebAppsLoader="loading; indicator: spinner"
+    (saveAttribute)="saveExpirationAttribute($event)"
+    [expirationAttribute]="expirationAttribute">
+  </app-expiration-settings>
+</ng-container>
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-expiration/vo-settings-expiration.component.html:10]: The `position-relative` class is needed for the correct displaying of the spinner (to cover just a defined HTML).
```suggestion
<div class="position-relative">
  <app-expiration-settings
    *perunWebAppsLoader="loading; indicator: spinner"
    (saveAttribute)="saveExpirationAttribute($event)"
    [expirationAttribute]="expirationAttribute">
  </app-expiration-settings>
</div>
```

**Review by HejdaJakub — COMMENTED** [review]
I'll look into it more deeply tomorrow, but here is my suggestion for now (I need to test it properly, if there aren't some side effects).

Also don't forget to implement changes also for a group.

**Review by xflord — COMMENTED** [review]
nothing to add, just the things mentioned by @HejdaJakub

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**